pkg/query-service/app/logs/v4/query_builder.go

```diff
@@ -29,3 +35,492 @@ const (
 	DISTRIBUTED_LOGS_V2_RESOURCE = "distributed_logs_v2_resource"
 	NANOSECOND                   = 1000000000
 )
+
+func getClickhouseLogsColumnDataType(columnDataType v3.AttributeKeyDataType) string {
+	if columnDataType == v3.AttributeKeyDataTypeFloat64 || columnDataType == v3.AttributeKeyDataTypeInt64 {
+		return "number"
+	}
+	if columnDataType == v3.AttributeKeyDataTypeBool {
+		return "bool"
+	}
+	return "string"
+}
+
+func getClickhouseKey(key v3.AttributeKey) string {
+	// check if it is a top level static field
+	if _, ok := constants.StaticFieldsLogsV3[key.Key]; ok && key.Type == v3.AttributeKeyTypeUnspecified {
+		return key.Key
+	}
```

Comment:
Is this needed _key.Type == v3.AttributeKeyTypeUnspecified_? Can there be an attribute which has same name as top level static field?

Reply:
Yes, there can be an attribute named `body`, `trace_id` etc. During enrichment, the type will be updated.